mono/tests/Makefile.am

```diff
@@ -2942,22 +2942,29 @@ EXTRA_DIST += test_lldb.py test-lldb.cs
 test-lldb: test-lldb.exe
 	python test_lldb.py $(JITTEST_PROG)
 
-noinst_LTLIBRARIES = libtest.la
+noinst_LTLIBRARIES = libtest.la libtest-c.la
 
 AM_CPPFLAGS = $(GLIB_CFLAGS)
 
+# libtest may be compiled as C++, but libtest-c may not be, due to empty structs.
+CFLAGS := $(filter-out @CXX_REMOVE_CFLAGS@, @CFLAGS@)
+
 if HOST_WIN32
 # gcc-3.4.4 emits incorrect code when making indirect calls to stdcall functions using a tail call
 # This shows up when compiling mono_test_marshal_delegate ()
-libtest_la_CFLAGS=-fno-optimize-sibling-calls
+libtest_la_CFLAGS=@CXX_ADD_CFLAGS@ -fno-optimize-sibling-calls
 # the exported names created by gcc for stdcall functions are missing the leading _, so MS.NET
 # can't find them. So we use --kill-at to remove the @ suffix as well.
 libtest_la_LDFLAGS=-no-undefined -rpath `pwd` -Wl,--kill-at
 else
+libtest_la_CFLAGS=@CXX_ADD_CFLAGS@
 libtest_la_LDFLAGS = -rpath `pwd`
 endif
 libtest_la_SOURCES = libtest.c
-libtest_la_LIBADD = $(top_builddir)/mono/eglib/libeglib.la
+libtest_la_LIBADD = $(top_builddir)/mono/eglib/libeglib.la libtest-c.la
+
+libtest_c_la_CFLAGS = @CXX_REMOVE_CFLAGS@
```

Comment:
I'm confused by the CFLAGS here: I thought `CXX_REMOVE_CFLAGS` was just a set of flags that might be present in a `CFLAGS` that must be dropped if we're adding the `CXX_ADD_CFLAGS`.  What does it mean to pass just the remove flags as the cflags for a library?

Reply:
oh, I guess you remove the `CXX_REMOVE_FLAGS` on line `2950`.  and so here you add them back? But what if one of the REMOVE_FLAGS wasn't present on the original CFLAGS?

Reply:
It is convoluted I agree. It evolved, and maybe can evolve again. I'm not sure.

CFLAGS contains flags that invalid for C++.
So we have to "remove" them.
And then "add" some flags to become C++.
But if some libraries within a directory remain  as C, which I did not exect, then we have to add back what we removed.

The expected usage model was just, as a directory is ported, add:
```CFLAGS := $(filter-out @CXX_REMOVE_CFLAGS@, @CFLAGS@) @CXX_ADD_CFLAGS@```

but directories that are a mix of C and C++ aren't so clean.

Reply:
Good question. Everything in REMOVE should be present in CFLAGS, as configure.ac computes them.